src/main/java/com/github/shiranr/scooters/db/Client.java

```diff
@@ -0,0 +1,89 @@
+package com.github.shiranr.scooters.db;
+
+import com.azure.cosmos.ConsistencyLevel;
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.CosmosContainer;
+import com.azure.cosmos.CosmosDatabase;
+import com.azure.cosmos.models.CosmosItemResponse;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.util.CosmosPagedIterable;
+import com.github.shiranr.scooters.domain.Scooter;
+
+import java.util.Optional;
+
+import static java.net.HttpURLConnection.HTTP_ACCEPTED;
+import static java.net.HttpURLConnection.HTTP_CREATED;
+import static java.net.HttpURLConnection.HTTP_OK;
+
+/**
+ * This class is singleton. The goal is to have a managed client for our service.
+ * Currently, this is not thread safe. Will be done later if needed.
+ * The DB HOST and PASSWORD are currently in ENV variables, should be moved to key vault.
+ */
+public class Client {
+    private static CosmosClient cosmos;
+    private static Client client;
+
+    static String DB_NAME = "lime";
+    static String CONTAINER_NAME = "scooters";
+
+    private Client() {
+        if (cosmos == null) {
+            String endpoint = System.getenv("DB_HOST");
+            String password = System.getenv("DB_PASSWORD");
+            cosmos = new CosmosClientBuilder()
+                    .endpoint(endpoint)
+                    .key(password)
+                    .consistencyLevel(ConsistencyLevel.EVENTUAL)
+                    .buildClient();
+        }
+    }
+
+    public static Client GetClient() {
```

Comment:
I would suggest to convert it into a DI based single instance.
It could be in a next iteration.
Otherwise, this pattern is very prawn to errors, especially with race conditions.

Reply:
Yes, there is also a comment this is currently not thread safe.
next iteration sync will be added.